src/libsyntax/diagnostic.rs

```diff
@@ -474,42 +474,56 @@ fn highlight_lines(err: &mut EmitterWriter,
         // how many digits must be indent past?
         while num > 0u { num /= 10u; digits += 1u; }
 
-        // indent past |name:## | and the 0-offset column location
-        let left = fm.name.len() + digits + lo.col.to_uint() + 3u;
         let mut s = String::new();
         // Skip is the number of characters we need to skip because they are
         // part of the 'filename:line ' part of the previous line.
-        let skip = fm.name.len() + digits + 3u;
+        let skip = fm.name.width(false) + digits + 3u;
```

Comment:
From the documentation: "Unicode Standard Annex #11 recommends that these characters be treated as 1 column (i.e., is_cjk = false) if the context cannot be reliably determined."

(It actually greatly varies on the font settings, so it cannot be reliably set by default.)